File: v2/src/main/java/com/api/v2/customers/services/CustomerRegistrationService.java
package com.api.v2.customers.services;

import com.api.v2.customers.responses.CustomerResponseDto;
import com.api.v2.people.requests.PersonRegistrationDto;
import org.springframework.http.ResponseEntity;
import reactor.core.publisher.Mono;

public interface CustomerRegistrationService {
    Mono<ResponseEntity<CustomerResponseDto>> register(PersonRegistrationDto registrationDto);
}

File: v2/src/main/kotlin/com/api/v2/customers/services/CustomerRegistrationService.kt
package com.api.v2.customers.services

import com.api.v2.customers.CustomerResponseDTO
import com.api.v2.people.requests.PersonRegistrationDTO
import org.springframework.http.ResponseEntity

interface CustomerRegistrationService {
    fun register(registrationDTO: PersonRegistrationDTO): ResponseEntity<CustomerResponseDTO>
}